config/passport.js

```diff
@@ -0,0 +1,79 @@
+const GoogleStrategy = require('passport-google-oauth').OAuth2Strategy;
+const LocalStrategy = require('passport-local').Strategy;
+const BearerStrategy = require('passport-http-bearer').Strategy;
+const configAuth = require('./settings');
+
+module.exports = function(passport, models) {
+
+	const localUser = models.user_account;
+
+	passport.serializeUser(function(user, done){
+		done(null, user);
+	});
+
+	passport.deserializeUser(function(user, done){
+		done(null, user);
+	});
+
+	passport.use(new GoogleStrategy({
+			clientID: configAuth.googleAuth.clientID,
+			clientSecret: configAuth.googleAuth.clientSecret,
+			callbackURL: configAuth.googleAuth.callbackURL,
+			access_type: configAuth.googleAuth.access_type
+		},
+	  	function(req, accessToken, refreshToken, profile, done) {
+	  		process.nextTick(function(){
+	  			localUser.findOrCreate({where: {
+	  				email: profile.emails[0].value,
+	  				provider: 'google'
+	  			}, defaults:{
+	  				email: profile.emails[0].value,
+	  				provider: 'google',
+	  				google_token: accessToken,
+	  				google_id: profile.id,
+	  				name: profile.displayName
+	  			}})
+	  			.spread((user, created)=> {
+	  				if(!created && !user){
+	  					return done(null, false, {'error': 'User with that email already exists'});
+	  				}
+	  				response = { email: user.email, name: user.name};
+	  				return done(null, response);
+	  			})
+	  			.catch(function(err){
+	  				return done(err);
+	  			});
+	  		});
+	  	}
+	));
+
+	passport.use('local-login', new LocalStrategy({
+		usernameField: 'email',
+		passwordField: 'password',
+		passReqToCallback: true
+		},
+		function(req, email, password, done){
+			process.nextTick(function(){
+			localUser.find({where: {
+				email: email
+			}}, {raw: true})
+			.then(data=> {
+				if(!data){
+					return done(null, false, {'error': 'Invalid email or password'});
+				}
+				if(data.provider=='google'){
+					return done(null, false, {'error': 'Please login with Google'});					
```

Comment:
The provider is an ENUM with 'local' and 'google' as the two values. So supposing a user has already signed in with google and tries to use this google id to sign in locally, this will prevent the user from doing that and will prompt him to sign in with google.